### ISSUE TITLE
Proper sorting only after Restack

### ISSUE THREAD

**Issue by Yanngo** [arkayenro/arkinventory#913]
Version: 30723

Some items are sorted into their categories when I log in but many of them are not properly sorted. Proper sorting kicks in only **after I press Restack** (one time or several times, doesn't work every time) or sometimes after I **change zones** by using a hearthstone or a teleport.

Maybe my sorting rules are too complicated. I don't know. If you want me to upload the saved settings somewhere, I can do that. Just tell me the website to upload to.

**Comment by Yanngo** [comment]
It is worth noting that once I open my bank, the items there stay unsorted for a fraction of the second but snap to their right positions immediately afterwards while items in my bags still remain unsorted.

**Comment by arkayenro** [comment]
do they sort when you click on the refresh button?

what sorting style is set for the bags via the profile?

**Comment by Yanngo** [comment]
> do they sort when you click on the refresh button?

Sometimes they fail to sort properly after Refresh. A low number of cases. They will get sorted eventually after several Refreshes with periods between them.

> what sorting style is set for the bags via the profile?

Quality. Item rarity, item (stat) level are at the top. Not sure I've made it or it is one of the default ones.

**Comment by arkayenro** [comment]
sorting style as in Instant / On Window Open / Manual

**Comment by Yanngo** [comment]
This setting is hard to find. On window open here.

**Comment by arkayenro** [comment]
yeah i know, im at work so cant tell what the option structure was to let you know exactly where

can you change it to Manual or whatever i called the last option (no instant or on window open) and see if that makes it work when you click on the refresh button

do get the translations loaded text when ai starts, or at least up to a minute later?

**Comment by Yanngo** [comment]
No idea about translations because I play with an EnGB client. Everything is English always. No messages about translations being loaded.
But a strange thing happened:
1. I switched to manual sorting.
2. Logged out.
3. Logged in and checked bags.
4. For the first time ever the bags were properly sorted on login.

**Comment by Jarhead101st** [comment]
i think this may be a wow problem rather than a arkinventory problem.

I noticed when i first log in and zone into my garrison (or other areas) it takes a few seconds to load everything. 

at the same time if i open my bags almost everything is in the default bag position as if it has not been assigned to any group. if i close and reopen my bags everything is assigned correctly. by this time all the things in the world are loaded too.

also if i try to hearth in these first few seconds before everything is loaded it fails and the tooltip shows "retrieving item information"

**Comment by arkayenro** [comment]
the sorting wont kick in until teh translations are loaded as it gets some of the text from tooltip requests so those need to resolve and once that happens it triggers an update event - so if you open the bag youll see everything in the default bag then a couple of seconds later it will re-sort and end up in the correct locations

you can disable the translations complete notification in the config under messages/alerts, you may want to check if that is on for the time being so you have a better idea of when AI is actually ready
